runtime/src/main/java/io/quarkiverse/quinoa/QuinoaDevProxyHandler.java

```diff
@@ -58,8 +67,73 @@ public void handle(final RoutingContext ctx) {
             next(currentClassLoader, ctx);
             return;
         }
-        final String uri = computeURI(resourcePath, request);
+
         final MultiMap headers = request.headers();
+        final String uri = computeURI(resourcePath, request);
+
+        // WebSocket connection for Hot Module Reloading (HMR)
+        if (!config.prodMode && headers.get("Connection").contains("Upgrade")) {
+            // Vite uses the root URL "/" which is special
+            final String forwardUri = request.uri().equals("/") ? "/" : uri;
```

Comment:
Are you sure this has no side-effect in other cases?

Reply:
It shouldn't its just this special case where Vite uses "/" but if I use your converted Uri its "/index.html" which then fowards to the wrong URL for the web socket.